finality-aleph/src/new_network/service.rs

```diff
@@ -29,16 +31,28 @@ pub struct IO<D: Data> {
     commands_from_manager: mpsc::UnboundedReceiver<ConnectionCommand>,
 }
 
-async fn run_peer_sender<N: Network, D: Data>(
+async fn run_peer_sender<N: Network<NS>, NS: NetworkSender, D: Data>(
```

Comment:
I think it might be more readable to put this inside of `Service` take the returned future and run that. No pressure if you don't like this, but it would eliminate one turbofish.

Reply:
I agree 100%